test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,376 @@
+        for (int i = tc->id * collections_per_thread;
+             i < (tc->id * collections_per_thread) + collections_per_thread && tc->running(); ++i) {
+            collection &coll = tc->db.get_collection(i);
+            scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name.c_str());
+            ccv.push_back({coll, std::move(cursor)});
+        }
+
+        std::string key;
+        uint64_t counter = 0;
+
+        while (tc->running()) {
+
+            auto &cc = ccv[counter];
+            tc->transaction.begin();
+
+            while (tc->transaction.active() && tc->running()) {
+
+                /* Generate a random key. */
+                key = random_generator::instance().generate_random_string(tc->key_size);
+
+                /* Insert a key value pair. */
+                if (tc->insert(cc.cursor, cc.coll.id, key)) {
+                    if (tc->transaction.can_commit())
+                        /* We are not checking the result of commit as it is not necessary. */
+                        tc->transaction.commit();
+                } else {
+                    tc->transaction.rollback();
+                }
+
+                /* Sleep the duration defined by the configuration. */
+                tc->sleep();
+            }
+
+            /* Rollback any transaction that could not commit before the end of the test. */
+            if (tc->transaction.active())
+                tc->transaction.rollback();
+
+            /* Reset our cursor to avoid pinning content. */
+            testutil_check(cc.cursor->reset(cc.cursor.get()));
+            if (++counter == collections_per_thread)
+                counter = 0;
+            testutil_assert(counter < collections_per_thread);
+        }
+    }
+
+    void
+    read_operation(test_harness::thread_context *tc) override final
+    {
+        /*
+         * Each read operation performs search_near calls with and without prefix enabled on random
+         * collections. Each prefix is randomly generated. The result of the seach_near call with
+         * prefix enabled is then validated using the search_near call without prefix enabled.
+         */
+        logger::log_msg(
+          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+        std::map<uint64_t, scoped_cursor> cursors;
+        // TODO - Should this be a fixed value ? Should it be from the configuration ?
+        const uint64_t prefix_size = 3;
+        const char *key_prefix, *key_default;
+        int exact_prefix, exact_default;
+        int ret;
+
+        while (tc->running()) {
+            /* Get a random collection to work on. */
+            collection &coll = tc->db.get_random_collection();
+
+            /* Find a cached cursor or create one if none exists. */
+            if (cursors.find(coll.id) == cursors.end()) {
+                cursors.emplace(
+                  coll.id, std::move(tc->session.open_scoped_cursor(coll.name.c_str())));
```

Comment:
Do we need `std::move` here? As far as I understand 'std::map::emplace' uses move constructor internally.

Reply:
Sorry, my bad. This code is good.